src/neuroconv/tools/spikeinterface/spikeinterface.py

```diff
@@ -1436,3 +1459,157 @@ def write_sorting(
+        for prop in qm.columns:
+            if prop not in sorting.get_property_keys():
+                sorting.set_property(prop, qm[prop])
+                tmp_properties.append(prop)
+    if waveform_extractor.is_extension("template_metrics"):
+        tm = waveform_extractor.load_extension("template_metrics").get_data()
+        for prop in tm.columns:
+            if prop not in sorting.get_property_keys():
+                sorting.set_property(prop, tm[prop])
+                tmp_properties.append(prop)
+
+    assert write_as in [
+        "units",
+        "processing",
+    ], f"Argument write_as ({write_as}) should be one of 'units' or 'processing'!"
+    if write_as == "units":
+        assert units_name == "units", "When writing to the nwbfile.units table, the name of the table must be 'units'!"
+    write_in_processing_module = False if write_as == "units" else True
+
+    with make_or_load_nwbfile(
+        nwbfile_path=nwbfile_path, nwbfile=nwbfile, metadata=metadata, overwrite=overwrite, verbose=verbose
+    ) as nwbfile_out:
+        if write_electrical_series:
+            if waveform_extractor._recording is None:
+                assert recording is not None, (
+                    "To write electrical series, the waveform_extractor needs to have a recording attached. "
+                    "For recordingless mode, you can use the 'recording' argument."
+                )
+            write_recording(waveform_extractor._recording, nwbfile_out, metadata, **write_electrical_series_kwargs)
+        else:
+            # we need add_electrodes_from_waveforms
+            if waveform_extractor._recording is None:
+                if recording is not None:
+                    assert isinstance(recording, BaseRecording), "'recording' needs to be a BaseRecording object."
+            else:
+                recording = waveform_extractor.recording
+            if recording is not None:
+                add_electrodes_info(recording, nwbfile=nwbfile_out, metadata=metadata)
+            else:
+                print(
+                    "To add the electrode table, the waveform_extractor needs to have a recording attached. "
+                    "For recordingless mode, you can use the 'recording' argument."
+                    "The electrode table will not be added."
+                )
+
+        add_units_table(
+            sorting=sorting,
+            nwbfile=nwbfile_out,
+            unit_ids=unit_ids,
+            property_descriptions=property_descriptions,
+            skip_properties=skip_properties,
+            write_in_processing_module=write_in_processing_module,
+            units_table_name=units_name,
+            unit_table_description=units_description,
+            write_waveforms=False,
+            waveform_means=template_means,
+            waveform_sds=template_stds,
+        )
+
+    # rm tmp properties
+    for prop in tmp_properties:
+        sorting.delete_property(prop)
```

Comment:
yes, that's a good idea

Reply:
Was the `clone` command not working as intended? Given we're already making this PR dependent on SI dev branch, it would be fine to wait on fixes for it here

Reply:
No, the problem is that clone reloads the data from the underlying files, so any in-memory changes would be lost. For example, if adding a property before calling the write function, that property would not be cloned. So temporary properties work better in this case